tests/www/test_views.py

```diff
@@ -2058,17 +2315,42 @@ def test_run_success_for_all_dag_user(self):
         self.check_content_in_response('', resp, resp_code=302)
 
     def test_blocked_success(self):
-        url = 'blocked'
         self.logout()
-        self.login()
+        username = 'blocked_success_user'
+        fab_utils.create_user(
+            self.app,
+            username=username,
+            role_name='blocked_success_role',
+            permissions=[
+                (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAGS),
+                (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
+                (permissions.ACTION_CAN_READ, permissions.RESOURCE_WEBSITE),
+            ]
+        )
+        self.login(username=username,
+                   password=username)
```

Comment:
This block appears a lot. I wonder if it's worth making a new helper methond on self:

```suggestion
        username = 'blocked_success_user'
        self.make_user_and_login
            username=username,
            permissions=[
                (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAGS),
                (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
                (permissions.ACTION_CAN_READ, permissions.RESOURCE_WEBSITE),
            ]
        )
```